maths/test/prime_factorization.test.ts

```diff
@@ -0,0 +1,26 @@
+import { factorize } from "../prime_factorization";
+
+
+interface TestCase {
+    n: number;
+    expected: Map<number, number>
+}
+
+const cases: TestCase[] = [
+    {n: 4, expected: new Map([[2, 2]])},
+    {n: 5, expected: new Map([[5, 1]])},
+    {n: 7, expected: new Map([[7, 1]])},
+    {n: 10, expected: new Map([[2, 1], [5, 1]])},
+    {n: 999, expected: new Map([[3, 3], [37, 1]])},
+    {n: 999999999999878, expected: new Map([[2, 1], [19, 1], [26315789473681, 1]])},
```

Comment:
I'm slightly worried about the runtime of this test. Could you choose a 100-1000x smaller number here?

Reply:
don't worry it is fast, I have tested it in my machine

Reply:
It is indeed not a problem yet, but such small slowdowns may add up in the future. But that can be addressed when/if it becomes a problem later on. For now I'm fine with it, hence the approving review.